paradox-nakadi-consumer-core/src/main/java/de/zalando/paradox/nakadi/consumer/core/http/handlers/AbstractEventsResponseBulkHandler.java

```diff
@@ -26,25 +25,20 @@
     @Override
     public void onResponse(final String content) {
         final String[] events = getEvents(content);
-        for (String event : events) {
-            final Optional<NakadiEventBatch<T>> optionalBatch = getEventBatch(event);
-            if (!optionalBatch.isPresent()) {
-                return;
-            }
+        for (final String event : events) {
+            final NakadiEventBatch<T> nakadiEventBatch = getEventBatch(event);
+
+            if (nakadiEventBatch != null) {
```

Comment:
[C] From what I have seen `nakadiEventBatch` cannot be null. How about `requireNonNull(getEventBatch(event))`?